components/ee/agent-smith/pkg/agent/metrics.go

```diff
@@ -41,6 +42,14 @@ func newAgentMetrics() *metrics {
 			Help:      "The total amount of failed attempts that agent-smith is trying to apply a penalty.",
 		}, []string{"penalty", "reason"},
 	)
+	m.egressViolations = prometheus.NewCounterVec(
```

Comment:
This is not a Gauge metric because we do not have a way to decrease the infringing workspace count.